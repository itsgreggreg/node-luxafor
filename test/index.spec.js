/* eslint-disable no-undef */
import Luxafor from '../src';

jest.mock('node-hid');

describe('Luxafor', () => {
  const setup = (...opts) => new Luxafor(...opts);

  it('Should use default pid and vid', () => {
    const instance = setup();

    expect({
      pid: instance.pid,
      vid: instance.vid
    }).toMatchSnapshot();
  });

  it('Should expose commands', () => {
    const instance = setup();

    expect(
      instance.commands
    ).toMatchSnapshot();
  });

  describe('write', () => {
    it('Should return the instance', () => {
      const instance = setup();

      expect(
        instance.write({})
      ).toMatchObject(instance);
    });

    it('Should write to the device', () => {
      const instance = setup();

      expect(
        instance.write({
          command: 'strobe',
          side: 'front',
          r: 255,
          g: 254,
          b: 253
        })
      ).toMatchSnapshot();
    });
  });

  describe('setColor', () => {
    it('Should return the instance', () => {
      const instance = setup();

      expect(
        instance.write({})
      ).toMatchObject(instance);
    });

    it('Write args', () => {
      const instance = setup();

      expect(
        instance.setColor(255, 254, 253)
      ).toMatchSnapshot();
    });
  });

});
