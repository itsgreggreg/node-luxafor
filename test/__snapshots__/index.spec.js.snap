// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Luxafor Should expose commands 1`] = `
Object {
  "color": 1,
  "fade": 2,
  "strobe": 3,
  "wave": 4,
}
`;

exports[`Luxafor Should use default pid and vid 1`] = `
Object {
  "pid": 62322,
  "vid": 1240,
}
`;

exports[`Luxafor setColor Write args 1`] = `
Luxafor {
  "commands": Object {
    "color": 1,
    "fade": 2,
    "strobe": 3,
    "wave": 4,
  },
  "device": Object {
    "write": [Function],
  },
  "patterns": Object {
    "police": 5,
  },
  "pid": 62322,
  "sides": Object {
    "back": 66,
    "both": 255,
    "front": 65,
  },
  "vid": 1240,
}
`;

exports[`Luxafor write Should write to the device 1`] = `
Luxafor {
  "commands": Object {
    "color": 1,
    "fade": 2,
    "strobe": 3,
    "wave": 4,
  },
  "device": Object {
    "write": [Function],
  },
  "patterns": Object {
    "police": 5,
  },
  "pid": 62322,
  "sides": Object {
    "back": 66,
    "both": 255,
    "front": 65,
  },
  "vid": 1240,
}
`;
